### PR TITLE
feat: add bulk export getCrawlerStatus lambda

### PR DESCRIPTION
**Pull request by carvantes**
Description of changes:

Another lambda function that'll be part of the bulk export state machine. 

This lambda function simply checks the status of the crawler that was started earlier by the `startCrawler` task and reports the status on its response. The state machine will use this status to decide the next step

Related PRs:
- https://github.com/awslabs/fhir-works-on-aws-deployment/pull/130

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/bulkExport/getCrawlerStatus.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            // This should never happen per the current Glue API specification
+            throw new Error(`Unknown crawler state: ${crawlerState}`);
+    }
+};
+
+export const getCrawlerStatusHandler: Handler<
+    BulkExportStateMachineGlobalParameters,
+    BulkExportStateMachineGlobalParameters
+> = async event => {
+    const { CRAWLER_NAME } = process.env;
```
Comment (rsmayda) [inline src/bulkExport/getCrawlerStatus.ts:45]: Is the crawler name always the same? just curious if this is the best way to pass in the parameter (env var or request param)

File: src/bulkExport/getCrawlerStatus.test.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+AWSMock.setSDKInstance(AWS);
+
+describe('getCrawlerStatusHandler', () => {
+    beforeEach(() => {
+        process.env.CRAWLER_NAME = 'crawlerName';
+        AWSMock.restore();
+    });
+
+    each([
```
Comment (rsmayda) [inline src/bulkExport/getCrawlerStatus.test.ts:19]: We should add the two error states:
- Unknown last crawl status: ${lastCrawlStatus}
- Unknown crawler state: ${crawlerState}

File: src/bulkExport/getCrawlerStatus.ts (hunk 1 above)
Comment (carvantes) [reply src/bulkExport/getCrawlerStatus.ts:45]: The crawler name is always the same. 
It is similar to how DDB table names are always the same and env variables are the preferred way to pass them to Lambda.

File: src/bulkExport/getCrawlerStatus.test.ts (hunk 2 above)
Comment (carvantes) [reply src/bulkExport/getCrawlerStatus.test.ts:19]: I did not test those errors on purpose.  

Those 2 code branches exist but they are unreachable unless Glue makes a breaking API change, which is extremely unlikely to happen. A test case like "when Glue does a breaking change on their response structure, then getCrawlerStatus should fail with \*error msg\*" isn't really useful.

How does that sound to you?

https://docs.aws.amazon.com/glue/latest/webapi/API_LastCrawlInfo.html#Glue-Type-LastCrawlInfo-Status
https://docs.aws.amazon.com/glue/latest/webapi/API_Crawler.html#Glue-Type-Crawler-State

Comment (nguyen102) [reply src/bulkExport/getCrawlerStatus.test.ts:19]: As a general rule of thumb, I generally add unit tests to cover all of the code path for the "system under test". In this case the "system under test" has a code path for throwing these "unknown" errors.

However, that is a good point that these edge cases are unlikely to be encountered during run time. 

Personally I would just cover the edge case since it's easy to do so, and as a general rule as stated above. However, this is not a blocking change for me.

**Review by nguyen102 — APPROVED** [review]

File: src/bulkExport/getCrawlerStatus.test.ts (hunk 2 above)
Comment (rsmayda) [reply src/bulkExport/getCrawlerStatus.test.ts:19]: Good to know context and the implausibility of the edge cases. I'd prefer to add them but not blocking for me either
